app/src/main/java/org/mozilla/fenix/settings/DefaultBrowserPreference.kt

```diff
@@ -34,4 +37,18 @@ class DefaultBrowserPreference @JvmOverloads constructor(
         val browsers = Browsers.all(context)
         switchView?.isChecked = browsers.isDefaultBrowser
     }
+
+    fun unsetOpenLinksInAPrivateTabIfNecessary() {
```

Comment:
@mcomella I ended up moving the shared method to DefaultBrowserPreference, but let me know if you think there is a better alternative

Reply:
Although I don't know Fenix that much, to me it makes sense to call this only when we unset ourselves as the default browser. However,  @mcomella  or @emalysz correct me if I'm wrong, what happens with the following use case: User kills browser -> goes in Settings/apps/default apps/ -> removes Fenix as default browser -> re-opens Fenix. Then we don't go through this loop as we start I think, right? I'm not too sure, which is why I'm asking 😛

Reply:
@MarcLeclair  I believe that use case is picked up by calling unsetOpenLinksInAPrivateTabIfNecessary (at least it did when I tested it), but it would be great if @mcomella could confirm :)

Reply:
@MarcLeclair afaict, the preference is used in two places:
- In the settings, to display the preference & provide the option to change it
- in IntentReceiverActivity, to use the preference, i.e. determine behavior based on it

This means we can update it:
- (current impl) Every time we open the app (so the most recent value is always available)
  - Caveat: Android doesn't have a single entry point so it's challenging to do this correctly
- (proposed impl) "just-in-time", right before we need to use it

Reply:
> I ended up moving the shared method to DefaultBrowserPreference, but let me know if you think there is a better alternative

I think we should move it somewhere else because `DefaultBrowserPreference` is a view: that means 1) we're initializing irrelevant code to call this method and 2) mixing our view and model layers (since the new code is at the model layer).

I think we could move it to `Settings`: there's some precedent there.